Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -350,19 +362,20 @@ void HcalDDDRecConstants::initialize(void) {
 					  << " at index " << i 
 					  << " of etaTable from SimConstant";
     } else {
+      etaTable.push_back(hpar->etaTable[ieta]);
       etaSimValu.push_back(std::pair<int,int>(ef,ieta));
     }
-    for (int k=0; k<hpar->etagroup[i]; ++k) ietaMap.push_back(i+1);
+    for (int k=0; k<(hpar->etagroup[i]); ++k) ietaMap.push_back(i+1);
     if (ieta <= hpar->etaMax[0]) ietaHB = i+1;
     if (ieta <= hpar->etaMin[1]) ietaHE = i+1;
   }
-  assert( hpar->etaMin[1] == ietaHE );
-  assert( hpar->etaMax[0] == ietaHB );
+  iEtaMin[1] = ietaHE;
```

Comment:
@bsunanda - according to your code it is not possible - both iEtaMax and iEtaMin are copies of hpar->etaMin and hpar->etaMax (see lines 348 and 349). They are not modified after that assignment.